### DISCUSSION TITLE
Configure span name extractor in java-http-client javaagent instrumentation

### DISCUSSION THREAD

**Discussion by ArtyomGabeev** [open-telemetry/opentelemetry-java-instrumentation#12139] [Q&A] [answered]
Hi everyone,

I'm using javaagent based instrumentation and want to configure in a single place spanNameExtractor, which is available only in library instrumentation: 
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/java-http-client/library/src/main/java/io/opentelemetry/instrumentation/httpclient/JavaHttpClientTelemetryBuilder.java#L97

Are there plans to allow to it for javaagent instrumentation e.g. by allowing to specify class name + reflection.

Or may be you can tell me, if it's possible to configure javaagent instrumentation via extensions:
https://opentelemetry.io/docs/zero-code/java/agent/extensions/

But looks like majority of factories are private ones.

Thanks,

**Comment by trask** [comment]
hi @ArtyomGabeev, can you achieve the span name modifications you need using a SpanProcessor?

**Reply by ArtyomGabeev** [reply]
Hi, @trask 

I was thinking to use SpanNameExtractor: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/SpanNameExtractor.java#L10

Is it possible to distinguish spans in SpanProcessor produced by specific instrumentation?
I'm particularly interested in java-http-client instrumentation.

If I understand correctly, I cannot access the original request URI in SpanProcessor.
What I'm try to achieve is provide better name for http spans to be "$METHOD $LOW_CARDINALITY_PATH"

**Reply by trask** [reply, marked as answer]
> Is it possible to distinguish spans in SpanProcessor produced by specific instrumentation?

yes, you can check the instrumentation scope name, e.g. `span.getInstrumentationScopeInfo().getName()`

> If I understand correctly, I cannot access the original request URI in SpanProcessor.

you can! check `span.getAttribute(UrlAttributes.URL_FULL)`

**Reply by ArtyomGabeev** [reply]
I did a quick look, and it should work, thank you!
